libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -999,13 +1164,71 @@ void WaveClip::Resample(int rate, BasicUI::ProgressDialog *progress)
    }
 }
 
+void WaveClip::ApplyStretchRatio(
+   const std::function<void(double)>& reportProgress)
+{
+   const auto stretchRatio = GetStretchRatio();
+   if (stretchRatio == 1.0)
+      return;
+
+   Finally Do { [this, trimLeftBeforeStretch = mTrimLeft,
+                 trimRightBeforeStretch = mTrimRight] {
+      this->mClipStretchRatio = 1.0;
+      this->mRawAudioTempo = this->mProjectTempo;
+      assert(this->GetStretchRatio() == 1);
+      this->SetTrimLeft(trimLeftBeforeStretch);
+      this->SetTrimRight(trimRightBeforeStretch);
```

Comment:
This isn't right, and not what I suggested.

This block is visited on function exit, whether normal or exceptional.

What I suggested is:
1. At top of the function, `bool success = false;`
2. At the very bottom of the "happy path", after all possibly throwing calls have not thrown, assign `success = true;`
3. In this Finally block, reassign the old trim left and trim right values, ONLY if `!success`
4. Also do the swapping, not assignment, of `mSequences`.  One swap below before the `Flush`, and another conditional swap here if not successful.  So, this also requires `newSequences` is declared before the `Finally`.